spec/app/domain/authentication/authn-azure/application_identity_spec.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe Authentication::AuthnAzure::ApplicationIdentity do
+  let(:subscription_id_annotation) { double("SubscriptionIdAnnotation") }
+  let(:subscription_id_annotation_empty) {double("SubscriptionIdAnnotationEmpty") }
+  let(:subscription_id_service_id_scoped_annotation) { double("SubscriptionIdServiceIdAnnotation") }
+
+  let(:good_service_id) { "MockService" }
+
+  let(:global_annotation_type) { "authn-azure" }
+  let(:granular_annotation_type) { "authn-azure/#{good_service_id}" }
+
+  def mock_annotation_builder(host_annotation_type, host_annotation_key, host_annotation_value)
+    allow(host_annotation_type).to receive(:values)
+                                     .and_return(host_annotation_type)
+    allow(host_annotation_type).to receive(:[])
+                                     .with(:name)
+                                     .and_return(host_annotation_key)
+    allow(host_annotation_type).to receive(:[])
+                                     .with(:value)
+                                     .and_return(host_annotation_value)
+  end
+
+  before(:each) do
+    mock_annotation_builder(subscription_id_annotation, "#{global_annotation_type}/subscription-id", "some-subscription-id-value")
+    mock_annotation_builder(subscription_id_annotation_empty, "#{global_annotation_type}/subscription-id", "")
+    mock_annotation_builder(subscription_id_service_id_scoped_annotation, "#{granular_annotation_type}/subscription-id", "some-subscription-id-service-id-scoped-value")
+  end
+
+  context "initialization" do
```

Comment:
i think that `context "An application identity in annotations"` will make more sense. Then you can remove the inner context with the same name.